torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1735,13 +1735,107 @@ int dictGetDefault(Stack& stack) {
   return 0;
 }
 
+// If the key is in the dict, return it. Else set it to the default value and
+// return that.
+int dictSetDefault(Stack& stack) {
+  auto default_value = pop(stack);
+  auto key = pop(stack);
+  auto dict = pop(stack).toGenericDict();
+  auto value = dict.find(key);
+  if (value == dict.end()) {
+    dict.insert(key, default_value);
+    push(stack, std::move(default_value));
+  } else {
+    push(stack, value->value());
+  }
+  return 0;
+}
+
+int dictPop(Stack& stack) {
+  auto key = pop(stack);
+  auto dict = pop(stack).toGenericDict();
+  auto value = dict.find(key);
+  if (value == dict.end()) {
+    AT_ERROR("KeyError: ", key);
+  } else {
+    auto erase_count = dict.erase(key);
+    TORCH_CHECK(
+        erase_count == 1, "Expected to erase 1 item, found ", erase_count);
+    push(stack, value->value());
+  }
+  return 0;
+}
+
+int dictPopDefault(Stack& stack) {
```

Comment:
can you unify these two functions into a single function? they almost share the same code except the empty case.